extensions/amp-action-macro/0.1/amp-action-macro.js

```diff
@@ -87,6 +100,28 @@ export class AmpActionMacro extends AMP.BaseElement {
     // We want the macro to be available wherever it is in the document.
     return true;
   }
+
+  /**
+   * Checks if the action is triggering itself.
+   * @param {!../../../src/service/action-impl.ActionInvocation} invocation
+   * @return {boolean}
+   * @private
+   */
+  isRecursiveInvocation_(invocation) {
+    return invocation.caller.getAttribute('id')
+        === this.element.getAttribute('id');
+  }
+
+  /**
+   * Checks if invoking action that was defined after it. This constraint is to
+   * prevent possible recursive calls.
+   * @param {number} invocationRegistrationId
+   * @return {boolean}
+   * @private
+   */
+  isInvalidMacroReference_(invocationRegistrationId) {
+    return invocationRegistrationId <= this.element['registrationId'];
```

Comment:
Can we use this instead? https://developer.mozilla.org/en-US/docs/Web/API/Node/compareDocumentPosition

Reply:
This is new to me. Fancy. done.